crates/proc-macro-api/src/msg.rs

```diff
@@ -19,8 +19,10 @@ pub const VERSION_CHECK_VERSION: u32 = 1;
 pub const ENCODE_CLOSE_SPAN_VERSION: u32 = 2;
 pub const HAS_GLOBAL_SPANS: u32 = 3;
 pub const RUST_ANALYZER_SPAN_SUPPORT: u32 = 4;
+/// Whether literals encode their kind as an additional u32 field and idents their rawness as a u32 field
+pub const EXTENDED_LEAF_DATA: u32 = 5;
```

Comment:
Headsup @vlad20012, this PR and https://github.com/rust-lang/rust-analyzer/pull/17601 make changes to the proc-macro server API adding a new version, adding two more u32s to literals on the wire, and one more u32 to idents. I have not really given this much thought tbf so please comment if you think we can do better than that. Its not set in stone until the next sync to rustc which will update the proc-macro server then.

Reply:
Note that syncs are currently on hold until we figure out the new `rustc_pattern_analysis` story.

Reply:
that's fine, probably better even as it gives me more time to test the proc-macro server :D

Reply:
@Veykril Thanks for mentioning! THe changes looks good to me